frontend/components/queries/queryResults/QueryResultsHeading/QueryResultsHeading.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+
+import Spinner from "components/Spinner";
+import Button from "components/buttons/Button";
+import TooltipWrapper from "components/TooltipWrapper";
+
+const pluraliseHost = (count: number) => {
+  return count > 1 ? "hosts" : "host";
+};
+
+const baseClass = "query-results-heading";
+
+interface IFInishButtonsProps {
+  onClickDone: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+  onClickRunAgain: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const FinishedButtons = ({
+  onClickDone,
+  onClickRunAgain,
+}: IFInishButtonsProps) => (
+  <div className={`${baseClass}__btn-wrapper`}>
+    <Button
+      className={`${baseClass}__done-btn`}
+      onClick={onClickDone}
+      variant="brand"
+    >
+      Done
+    </Button>
+    <Button
+      className={`${baseClass}__run-btn`}
+      onClick={onClickRunAgain}
+      variant="blue-green"
+    >
+      Run again
+    </Button>
+  </div>
+);
+
+interface IStopQueryButtonProps {
+  onClickStop: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const StopQueryButton = ({ onClickStop }: IStopQueryButtonProps) => (
+  <div className={`${baseClass}__btn-wrapper`}>
+    <Button
+      className={`${baseClass}__stop-btn`}
+      onClick={onClickStop}
+      variant="alert"
+    >
+      <>Stop</>
+    </Button>
+  </div>
+);
+
+interface IQuertResultsHeadingProps {
+  respondedHosts: number;
+  targetsTotalCount: number;
+  isQueryFinished: boolean;
+  onClickDone: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+  onClickRunAgain: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+  onClickStop: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+const QuertResultsHeading = ({
+  respondedHosts,
+  targetsTotalCount,
+  isQueryFinished,
+  onClickDone,
+  onClickRunAgain,
+  onClickStop,
+}: IQuertResultsHeadingProps) => {
+  const percentResponded =
+    targetsTotalCount > 0
+      ? Math.round((respondedHosts / targetsTotalCount) * 100)
+      : 0;
+
+  const PAGE_TITLES = {
+    RUNNING: `Querying selected ${pluraliseHost(targetsTotalCount)}`,
+    FINISHED: "Query finished",
+  };
+
+  const pageTitle = isQueryFinished
+    ? PAGE_TITLES.FINISHED
+    : PAGE_TITLES.RUNNING;
+
+  return (
+    <div className={`${baseClass}`}>
+      <h1>{pageTitle}</h1>
+      <div className={`${baseClass}__targeted-wrapper`}>
+        <span className={`${baseClass}__targeted-count`}>
+          {targetsTotalCount}
+        </span>
+        <span>&nbsp;{pluraliseHost(targetsTotalCount)} targeted.</span>
```

Comment:
American English spelling with a Z instead of a S?

Reply:
lol i need to find a vs code extension for this.